micro-rdk/src/common/data_manager.rs

```diff
@@ -369,10 +445,16 @@ where
                     if next_message.is_empty()
                         || ((next_message.len() + current_chunk_size) > MAX_SENSOR_CONTENTS_SIZE)
                     {
-                        let data: Result<Vec<SensorData>, prost::DecodeError> =
-                            current_chunk.drain(..).map(SensorData::decode).collect();
+                        let data: Result<Vec<SensorData>, DataSyncError> = current_chunk
+                            .drain(..)
+                            .map(create_time_corrected_reading(self.robot_start_time))
```

Comment:
Unless rust is able to see this and hoist this call automatically, this will create a new time adjuster every time in the loop, and then probably has to do dynamic dispatch under the `Fn`, but as far as I can see `robot_start_time` never changes. Do you even need it anymore? Could it just be a regular function on `self`? At minimum I think you could hoist the creation of the adjuster outside the loop.